src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -47,101 +40,110 @@ const VideoInputBackgroundBlurControl: React.FC<Props> = ({
-              );
-            }
-            setIsLoading(false);
+    const handleClick = async (deviceId: string): Promise<void> => {
+      try {
+        // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+        // If background blur is not on, then do a normal video selection
+        if (isVideoTransformDevice(selectedDevice) && !isLoading) {
+          setIsLoading(true);
+          if ('chooseNewInnerDevice' in selectedDevice) {
+            const transformedDevice =
+              // @ts-ignore
+              selectedDevice.chooseNewInnerDevice(deviceId);
+            await selectVideoinput(transformedDevice);
           } else {
-            await meetingManager.selectVideoInputDevice(option.deviceId);
+            console.error('Transform device cannot choose new inner device');
           }
-        }}
-      />
-    ));
-    if (isBackgroundBlurSupported) {
-      const videoTransformOptions: ReactNode = (
-        <PopOverItem
-          key="videoinput"
-          children={
+          setIsLoading(false);
+        } else {
+          await selectVideoinput(deviceId);
+        }
+      } catch (error) {
+        console.error('Failed to select video input device');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    const getDropdownWithVideoTransformOptions = async (): Promise<void> => {
+      const deviceOptions: ReactNode[] = await Promise.all(
+        videoDevices.map(async (device) => (
+          <PopOverItem
+            key={device.deviceId}
+            checked={await isOptionActive(selectedDevice, device.deviceId)}
+            onClick={async (): Promise<void> => handleClick(device.deviceId)}
+          >
+            <span>{device.label}</span>
+          </PopOverItem>
+        ))
+      );
+
+      if (isBackgroundBlurSupported) {
+        const videoTransformOptions: ReactNode = (
+          <PopOverItem
+            key="videoinput"
+            checked={isVideoTransformDevice(selectedDevice)}
+            disabled={isLoading}
+            onClick={toggleBackgroundBlur}
+          >
             <>
               {isLoading && <Spinner width="1.5rem" height="1.5rem" />}
               {backgroundBlurLabel}
             </>
-          }
-          checked={isVideoTransformDevice(activeVideoDevice)}
-          disabled={isLoading}
-          onClick={toggleBackgroundBlur}
-        />
-      );
-      deviceOptions.push(<PopOverSeparator key="separator" />);
-      deviceOptions.push(videoTransformOptions);
-    }
-    setDropdownWithVideoTransformOptions(deviceOptions);
+          </PopOverItem>
+        );
+        deviceOptions.push(<PopOverSeparator key="separator" />);
+        deviceOptions.push(videoTransformOptions);
+      }
+      setDropdownWithVideoTransformOptions(deviceOptions);
+    };
+
+    getDropdownWithVideoTransformOptions();
   }, [
     createBackgroundBlurDevice,
-    activeVideoDevice,
+    selectVideoinput,
```

Comment:
typo: selectVideoInput?

Reply:
It is `selectVideo` input.
```js
const selectVideoinput = useSelectVideoInputDevice();
```

Will remove these `useSelect*Device` hooks and updated to `meetingManager.start*Device` instead in a later PR.

Reply:
Updated.